src/main/java/com/minecolonies/coremod/items/ItemCactusDoor.java

```diff
@@ -0,0 +1,99 @@
+    //TODO REMOVE -  private final String name;
+
+    ItemCactusDoor(final Block block, final String name)
+    {
+        super();
+        this.block = block;
+        setRegistryName(name);
+        super.setUnlocalizedName(Constants.MOD_ID.toLowerCase() + "." + name);
+        this.setCreativeTab(ModCreativeTabs.MINECOLONIES);
+
+    }
+
+    @Override
+    public EnumActionResult onItemUse(EntityPlayer player, World worldIn, BlockPos pos, EnumHand hand, EnumFacing facing, float hitX, float hitY, float hitZ)
+    {
+        if (facing != EnumFacing.UP)
+        {
+            return EnumActionResult.FAIL;
+        }
+        else
+            {
+            final IBlockState iblockstate = worldIn.getBlockState(pos);
+            final Block blockIn = iblockstate.getBlock();
+            if (!blockIn.isReplaceable(worldIn, pos))
+            {
+                pos = pos.offset(facing);
+            }
+
+            final ItemStack itemstack = player.getHeldItem(hand);
+            if (player.canPlayerEdit(pos, facing, itemstack) && this.block.canPlaceBlockAt(worldIn, pos))
+            {
+                EnumFacing enumfacing = EnumFacing.fromAngle((double)player.rotationYaw);
+                final int i = enumfacing.getFrontOffsetX();
+                final int j = enumfacing.getFrontOffsetZ();
+                final boolean flag = (i < 0 && hitZ < 0.5F || i > 0 && hitZ > 0.5F || j < 0 && hitX > 0.5F || j > 0 && hitX < 0.5F);
+                placeDoor(worldIn, pos, enumfacing, this.block, flag);
+                SoundType soundtype = worldIn.getBlockState(pos).getBlock().getSoundType(worldIn.getBlockState(pos), worldIn, pos, player);
+                worldIn.playSound(player, pos, soundtype.getPlaceSound(), SoundCategory.BLOCKS, (soundtype.getVolume() + 1.0F) / 2.0F, soundtype.getPitch() * 0.8F);
+                itemstack.shrink(1);
+                return EnumActionResult.SUCCESS;
+            }
+            else
+                {
+                return EnumActionResult.FAIL;
+            }
+        }
+    }
+
+    public static void placeDoor(World worldIn, BlockPos pos, EnumFacing facing, Block door, boolean isRightHinge)
+    {
+        final BlockPos blockpos = pos.offset(facing.rotateY());
+        final BlockPos blockpos1 = pos.offset(facing.rotateYCCW());
+        final int i = (worldIn.getBlockState(blockpos1).isNormalCube() ? 1 : 0) + (worldIn.getBlockState(blockpos1.up()).isNormalCube() ? 1 : 0);
+        final int j = (worldIn.getBlockState(blockpos).isNormalCube() ? 1 : 0) + (worldIn.getBlockState(blockpos.up()).isNormalCube() ? 1 : 0);
+        final boolean flag = worldIn.getBlockState(blockpos1).getBlock() == door || worldIn.getBlockState(blockpos1.up()).getBlock() == door;
+        final boolean flag1 = worldIn.getBlockState(blockpos).getBlock() == door || worldIn.getBlockState(blockpos.up()).getBlock() == door;
+        if ((!flag || flag1) && j <= i)
+        {
+            if (flag1 && !flag || j < i)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)